### PR TITLE
2363 allows to $set variable to filter output as string

### LINKED ISSUE #2363: Extend <$set> widget to allow saving filter results as a single entry
The `<$set>` widget generates a list as output when used with the `filter` attribute. That has the side effect that the output is formatted according to standard TW list rules: space separation, and double square brackets to quote titles containing spaces.

Thus, a filter that outputs a single result such as `Filter Result` will be formatted as `[[Filter Result]]` in the output.

This can be fixed by adding a means to save just the first entry in the list as a single title, without any of the quoting.

[ref discussion](https://groups.google.com/forum/#!topic/tiddlywiki/gEbutPLs1vs)

### PR DESCRIPTION
**Pull request by tobibeer**
fixes #2363

demo: http://2363.tiddlyspot.com

### CODE REVIEW

**Comment by inmysocks** [comment]
I haven't had a chance to look at the code yet, but I would really appreciate it if this is added to the core.

**Comment by Jermolene** [comment]
@tobibeer your example is:

```
<$set name="myVariable" filter="[[Customize TiddlyWiki]]" plain>
<$text text=<<myVariable>>/>
</$set>
```

I'm more inclined to make the functionality be a filter operator:

```
<$set name="myVariable" filter="[[Customize TiddlyWiki]split[ ]]">
<$text text=<<myVariable>>/>
</$set>
```

Only we can't use `split` because it's already taken!

**Comment by tobibeer** [comment]
You can use tobibeer.github.io/tw5-plugins#split as follows:

```
<$set name="words" filter="[[foo bar baz]split[ ]]">
<<words>>
</$set>
```

You can try it out on the linked site.

**Comment by twMat** [comment]
@Jermolene - what was the conclusion here; to use @tobibeer 's split or to make a new operator titled something else?

And was something even implemented+merged so #2363 can be closed now?

**Comment by tobibeer** [comment]
It appears, it was decided to introduce another attribute instead, namely **select** (https://github.com/Jermolene/TiddlyWiki5/commit/b35544bf4946596e60f7b0c7f0a55fa490ac2a24), see my comment at https://github.com/Jermolene/TiddlyWiki5/commit/b35544bf4946596e60f7b0c7f0a55fa490ac2a24#commitcomment-19500501.

I'm not sure it is intuitive that **select** would not return the item in a stringifyList manner, but that's what it does now.

**Comment by Jermolene** [comment]
Hi @tobibeer here's the rationale for going with the implementation I originally suggested:

* The textual representation of the value of a tiddler field is a string with no special quoting
* The textual representation of a list of items is to quote any items containing spaces, and join the items with a space

Thus, `<$set name=foo filter=bar/>` returns the list as a string with the quoting behaviour, because that's how lists are converted to strings. Meanwhile, `<$set name=foo filter=bar select="0"/>` selects an item from the list, and the string representation of an item is unquoted, so the result is unquoted.

In other words, it's not a matter of choosing whether you want the result quoted with stringifyList, it's a matter of whether the result you choose is a list or a single item.

**Comment by tobibeer** [comment]
> In other words, it's not a matter of choosing whether you want the result quoted with stringifyList, it's a matter of whether the result you choose is a list or a single item.

I'm happy to see the update and that it actually allows to do more than a simple "plain" would have. 

Anyway, if we want to select the last one, we can always add **last[]** to the filter expression or any other method that does some more sophisticated selecting on the filter results.
